frontend/app/routes/protected/person-case/primary-docs.tsx

```diff
@@ -44,45 +52,166 @@ export async function action({ context, request }: Route.ActionArgs) {
   const t = await getFixedT(request, handle.i18nNamespace);
 
   const formData = await request.formData();
+  const action = formData.get('action');
 
-  if (formData.get('action') === 'back') {
-    throw i18nRedirect('routes/protected/person-case/privacy-statement.tsx', request); //TODO: change it to redirect to file="routes/protected/person-case/request-details.tsx"
-  }
+  switch (action) {
+    case 'back': {
+      throw i18nRedirect('routes/protected/index.tsx', request);
+    }
 
-  // submit action
-  const schema = v.object({
-    currentStatusInCanada: v.pipe(
-      v.string(t('protected:primary-identity-document.current-status-in-canada.required')),
-      v.trim(),
-      v.nonEmpty(t('protected:primary-identity-document.current-status-in-canada.required')),
-      v.literal(
-        'canadian-citizen-born-outside-canada',
-        t('protected:primary-identity-document.current-status-in-canada.invalid'),
-      ),
-    ),
-  });
+    case 'next': {
+      const input: { currentStatusInCanada?: string; documentType?: string } = {};
 
-  const input = { currentStatusInCanada: formData.get('currentStatusInCanada') as string };
-  const parsedDataResult = v.safeParse(schema, input, { lang });
+      const currentStatusSchema = v.object({
+        currentStatusInCanada: v.pipe(
+          v.string(t('protected:primary-identity-document.current-status-in-canada.required')),
+          v.trim(),
+          v.nonEmpty(t('protected:primary-identity-document.current-status-in-canada.required')),
+          v.picklist(VALID_CURRENT_STATUS, t('protected:primary-identity-document.current-status-in-canada.invalid')),
+        ),
```

Comment:
Validate with `picklist`.

```suggestion
        currentStatusInCanada: v.picklist(VALID_CURRENT_STATUS, t('protected:primary-identity-document.current-status-in-canada.required')),
```